Mockup of possible oghliner site design

darkwing opened the pull request:
This is a very, very rough idea for what an independent oghliner site could look like.  Very unrefined, code isn't great, but  maybe it can serve as a starting point.

mykmelez commented:
Hey, that's pretty! I like the style, it looks Mozilla-y. And feels more approachable and readable.

The only noteworthy change I would make is to the "Download Oghliner" button. Those buttons are pretty common on project pages hosted on GitHub Pages, but the "download" metaphor doesn't make much sense for this project.

For (prospective) users, I would direct them to documentation about installing Oghliner via NPM. For (prospective) contributors, I would encourage them to either "fork" the repo (a la the common "fork me on GitHub" ribbon, although it doesn't have to be a ribbon) or "browse" the repo.

mykmelez commented on app/styles/stylesheet.css:
```diff
@@ -48,33 +41,86 @@ table {
+}
+
+#tabzilla a {
+  background-image: url("https://www.mozilla.org/media/img/tabzilla/tabzilla-static-high-res.c4e717b903f2.png");
```
Is this URL stable? I wonder if we should copy the file into our project, given what looks like a hash in the filename, which presumably will change if the file is ever updated.

mykmelez commented on app/styles/stylesheet.css:
```diff
@@ -48,33 +41,86 @@ table {
+#tabzilla {
+  position: relative;
+  float: right;
+}
```
It's too bad there doesn't seem to be a stylesheet just for tabzilla that we could include.

mykmelez commented on app/styles/stylesheet.css:
```diff
@@ -48,33 +41,86 @@ table {
+  -webkit-background-size: 147px 37px;
+  background-size: 147px 37px;
+
+      position: relative;
```
Nit: there's some funky indentation here.

darkwing commented:
@mykmelez I've addressed each of your comments.  I've also:
1.  Added Tabzilla as a submodule so we can get updates from the official repo.
2.  Added fonts per Tabzilla's prescription
3.  Added button styles for NPM and GitHub

I've not added a CSS preprocessor like Stylus to keep things simple.  If you prefer I did, please let me know.

mykmelez commented:
> Added Tabzilla as a submodule so we can get updates from the official repo.

It looks like Tabzilla is now available as [an NPM module](https://www.npmjs.com/package/mozilla-tabzilla). Since we already use NPM to install other dependencies, can we use it here too? That'd simplify dependency management.

(I'm not opposed to Git submodules, and have used them happily in the past. Just wondering about it in this particular case, given the availability of the NPM module.)

darkwing commented:
Updated to use tabzilla NPM instead of submodule!

mykmelez commented on gulpfile.js:
```diff
@@ -25,10 +25,14 @@ var oghliner = require('./index.js');
 });
 
+gulp.task('build-tabzilla', function() {
+  return gulp.src('node_modules/mozilla-tabzilla/**').pipe(gulp.dest('app/styles/tabzilla'));
```
Copying the whole directory ends up copying (and caching) some files that aren't necessary:

> [13:44:33] Caching static resource "dist/styles/tabzilla/LICENSE" (16.73 kB)
> [13:44:33] Caching static resource "dist/styles/tabzilla/README.md" (4.75 kB)
> [13:44:33] Caching static resource "dist/styles/tabzilla/css/_tabzilla.scss" (846 B)
> [13:44:33] Caching static resource "dist/styles/tabzilla/css/tabzilla.css" (846 B)
> [13:44:33] Caching static resource "dist/styles/tabzilla/index.js" (424 B)
> [13:44:33] Caching static resource "dist/styles/tabzilla/media/img/tabzilla-static-high-res.png" (1.21 kB)
> [13:44:33] Caching static resource "dist/styles/tabzilla/media/img/tabzilla-static.png" (2.23 kB)
> [13:44:33] Caching static resource "dist/styles/tabzilla/package.json" (1.48 kB)

So I would copy just the tabzilla.css file and the image files. That's a bit more brittle, since the required files and their filenames could change in the future. But it reduces the space required to store the files on the server and clients, and it makes the offlining process less brittle, since a failure to offline a single resource can cause the entire process to fail.

mykmelez commented on app/index.html:
```diff
@@ -5,44 +5,71 @@
+
+      <section class="buttons">
+        <a href="" class="button">Fork on GitHub</a>
+        <a href="" class="button">Discover on npm</a>
```
It looks like these buttons are missing references.

darkwing commented:
Ugh.  Updates made!

mykmelez commented:
Sweet.
